cpp/src/IceSSL/OpenSSLEngine.h

```diff
@@ -26,24 +26,14 @@ namespace IceSSL
             IceInternal::TransceiverPtr
             createTransceiver(const IceSSL::InstancePtr&, const IceInternal::TransceiverPtr&, const std::string&, bool)
                 final;
-
-#ifndef OPENSSL_NO_DH
-            DH* dhParams(int);
-#endif
-
             SSL_CTX* context() const;
             void context(SSL_CTX*);
             std::string sslErrors() const;
+            std::string password() const { return _password; }
 
         private:
-            void cleanup();
-            SSL_METHOD* getMethod();
-
             SSL_CTX* _ctx;
-
-#ifndef OPENSSL_NO_DH
-            IceSSL::OpenSSL::DHParamsPtr _dhParams;
-#endif
+            std::string _password;
```

Comment:
Is it necessary to keep the password (presumably in clear text) as a field of this class?

Reply:
We can query `properties` in `password()` implementation. But security wise is the same. The properties dictionary also contains a copy of the password in clear text.